torch/_inductor/graph.py

```diff
@@ -1149,6 +1183,29 @@ def codegen(self):
         self.scheduler.codegen()
         return self.wrapper_code.generate(self.is_inference)
 
+    def codegen_subgraph(self, parent_wrapper_code):
+        """
+        This is a more compact version of the `codegen()` above
+        where we codegen this graph as a subgraph of some parent
+        graph. The parent graph's wrapper is passed as an
+        argument: the intention is to inline codegening of the
+        subgraph in the parent's wrapper code (including the
+        generated kerenls). The wrapper code is not finalized
+        (via `.generate()` call), as this will be done in the
+        parent graph's `codegen()`.
+        """
+        from .scheduler import Scheduler
+
+        old_wrapper_code = self.wrapper_code
```

Comment:
surprised this is all we need.... but im sure more will get uncovered with further PRs

Reply:
+ add test first

Reply:
This seems sufficient in https://github.com/pytorch/pytorch/pull/119672, but yeah, may be more cases requiring extension in the future.